src/Company.re

```diff
@@ -1,8 +1,6 @@
-let nullable = Faker.nullable;
-
 type t = {
   suffixes: unit => Js.Array.t(string),
-  companyName: Js.Nullable.t(int) => string,
+  companyName: int => string,
```

Comment:
So, thanks for checking through this code. I am trying to go through what happened here. Basically I was leaning hard on the tests. Since they all passed, I figured we were good. I don't really understand test construction yet so I am wondering if you can tell me what error in thinking I have made that you are keying in on. Obviously, you are seeing something obvious that the tests are not. What is going on here? Thanks in advance for the lesson, @Schniz .

Reply:
no problems, thanks for contributing :) 

The idea here is that tests don't need to change, because we already had a Reason/OCaml wrapper around Faker that transformed every `option('a)` to `Js.Nullable.t('a)`. So test files shouldn't be changed. The fact that we now need to handle `option`s using `Belt.Option.getExn` show that our Reason API changed which shouldn't happen in this PR.

Optional arguments have weird syntax to them, because of the curry-by-default nature of OCaml: https://reasonml.github.io/docs/en/function#optional-labeled-arguments

you can use labeled arguments and make them optional by providing a positional `unit` argument like so:

```reason
let my_fn = (~optional_arg=?, ()) => { /* ... */ };

my_fn(); // works!
my_fn(~optional_arg="hello!", ()); // pass the optional argument
my_fn(()); // works, this is the same as `my_fn();`
```

If you'll try to do something like the following:

```reason
let my_fn = (~optional_arg=?) => { /* ... */ };
```

[You'll get a warning](https://reasonml.github.io/en/try?rrjsx=true&reason=DYUwLgBAtgng+gMwHYQLwQBQD8D2AHMASxyQENg5SAnAc1QH4BKNAPggG8IB6AKggDpBEHlwgBfANxA), saying:

```
This optional parameter in final position will, in practice, not be optional.
Reorder the parameters so that at least one non-optional one is in final position or, if all parameters are optional, insert a final ().

Explanation: If the final parameter is optional, it'd be unclear whether a function application that omits it should be considered fully applied, or partially applied. Imagine writing `let title = display("hello!")`, only to realize `title` isn't your desired result, but a curried call that takes a final optional argument, e.g. `~showDate`.

Formal rule: an optional argument is considered intentionally omitted when the 1st positional (i.e. neither labeled nor optional) argument defined after it is passed in.
```

tl;dr - writing in Reason:

```reason
let my_fn = (~optional_arg=?, ()) => { /* ... */ };
```

is practically the same as [writing the following in TypeScript](https://www.typescriptlang.org/play/?ssl=5&ssc=3&pln=3&pc=1#code/C4TwDgpgBAzg9gWwgfVJKBeKA7ArggIwgCcBuAKHIBsJgoERkAzbTKACgG8o4xgBLONgCGVZMOIBzKAF8AXFG68BQ0eKkB+BfCSpw0GW04yAlJgB8i8lCgB6W1AB0z8jNJA):

```typescript
let my_fn = ({ optional_arg }: { optional_arg?: some_type } = {}) => {
  // ...
};
```

Reply:
Wow. This is an excellent explanation. Thank you for taking the time to write this.